### PR TITLE
KIT-231 - Display for rss feeds

### PR DESCRIPTION
**Pull request by Tylyfox**
### Desktop design
image
### Responsive design for mobile
image
 All Rss feed are sorting by publication date.

### CODE REVIEW

**Review by emericb — APPROVED** [review]

File: src/app/components/rss/rss-feed-all/rss-feed-all.component.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          this.sortRssDataItemsByDate();
+        },
+        error: (error: any): void => {
+          console.error('Erreur lors de la récupération du flux RSS', error);
+        },
+      });
+    });
+  }
+
+  //Add feed title to each list item
```
Comment (Benja-san) [inline src/app/components/rss/rss-feed-all/rss-feed-all.component.ts:35]: remove dead code

File: src/app/components/rss/rss-feed-all/rss-feed-all.component.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  @Input() rssLink!: string[];
+  rssDataItems?: any[] = [];
+
+  constructor(private rssFeedService: RssFeedService) {}
+
+  ngOnInit(): void {
+    this.getRssData();
+  }
+
+  getRssData(): void {
```
Comment (Benja-san) [inline src/app/components/rss/rss-feed-all/rss-feed-all.component.ts:19]: Good job! It is a bit imperative however, if you need to use this subscribing method in another file you'll have to duplicate code.
If you don't need it anywhere else, keep it like that, else transfer all the data transformation and sorting into the service in order to just give the method to your component

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/components/rss/rss-feed-all/rss-feed-all.component.ts (hunk 2 above)
Comment (Tylyfox) [reply src/app/components/rss/rss-feed-all/rss-feed-all.component.ts:19]: the two functions of title In item and sort asc can be reused in the display of rss feeds specific to a feed or for later when filtering. And I modified the api link to also provide filtering.

**Review by emericb — DISMISSED** [review]

File: src/environments/environment.development.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const environment = {
+  keyApi: 'ocgvg1llehvfjejyglkyv9xjx9uqb9nfgtntlay3',
```
Comment (Benja-san) [inline src/environments/environment.development.ts:2]: you forgot to put this file on the gitignore IMPORTANT

File: src/environments/environment.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const environment = {
+  keyApi: 'ocgvg1llehvfjejyglkyv9xjx9uqb9nfgtntlay3',
```
Comment (Benja-san) [inline src/environments/environment.ts:2]: PUT a generic false apikey to make an example for environment.development.ts

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — DISMISSED** [review]

**Review by emericb — DISMISSED** [review]

**Review by emericb — APPROVED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**